Custom-ESLint-checks.md

```diff
@@ -0,0 +1,329 @@
+
+```js
+export default function(context) {
+  return {
+    Literal: function(node) {
+      if (node.value === "Hello, world!") {
+        context.report({
+          node,
+          message: 'Do not use "Hello, world!" strings',
+        });
+      }
+    }
+  };
+};
+```
+
+Notice that the node we are examining gets passed into our function and that we report errors by calling `context.report`. In the lower right on AST Explorer, you can see that this lint rule works as expected:
+
+![Screenshot of AST Explorer with the lint rule running on code with two "Hello, world!" literals and one constructed string. The linter output flags only the literals.](images/customEslintChecks/helloWorldRule.png)
+
+#### Add the rule to Oppia
+
+To add the rule to Oppia's suite of lint checks, we need to add some more metadata describing each problem the rule can find. For our hello world rule, we would create a JavaScript file in `scripts/linters/custom_eslint_checks/rules/hello-world.js` with the following code:
+
+```js
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: '"Hello, world!" strings should not be used',
+      category: 'Best Practices',
+      recommended: true,
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      helloWorld: 'Do not use "Hello, world!" strings',
+    },
+  },
+
+  create: function(context) {
+    return {
+      Literal: function(node) {
+        if (node.value === "Hello, world!") {
+          context.report({
+            node: node,
+            messageId: 'helloWorld',
+          });
+        }
+      }
+    };
+  }
+};
+```
+
+First, consider the `meta` key at the top of the exported object. The associated object contains the rule's metadata. Under `docs` we describe the rule, and under `messages` we list the error messages that this rule can raise. ESLint supports rules that suggest fixes, but since we can't automatically fix a `Hello, world!` usage, we set `fixable` to `null`.
+
+Next, consider the `create` key. This key maps to the function we wrote in AST Explorer. However, notice that the values passed to `context.report()` have changed. Instead of specifying the error message directly, it's better to reference a message from the rule's metadata.
+
+We also need to add our rule to `.eslintrc`. Add `"oppia/hello-world": "error",` under the top-level `rules` key.
+
+#### Run the linter with the new rule
+
+To run the linter with your new rule, follow these steps:
+
+1. Delete the `node_modules/eslint-plugin-oppia/` directory so that in the next steps, the lint rules get re-installed.
+
+2. Run `yarn install` to install the custom rules, including the one you just wrote.
```

Comment:
Should this be from the oppia/ dir? If so, please specify that.

Reply:
done